WooCommerce/Classes/ViewRelated/Orders/Order Details/OrderFulfillmentNoticePresenter.swift

```diff
@@ -0,0 +1,70 @@
+import Foundation
+import Combine
+
+/// Presents notices for order fulfillment and handling of undos and retries.
+///
+/// This is the UI counterpart for `OrderFulfillmentUseCase`.
+///
+/// This class is meant to live longer than the `ViewController` that creates it. So a lot of the
+/// calls in here will capture `self`. The reason is we want to notify users if something fails in
+/// `OrderFulfillmentUseCase`. And we want to do this even if the user has already left the
+/// `ViewController` (i.e. `OrderDetailsViewController`).
+///
+final class OrderFulfillmentNoticePresenter {
+
+    private let noticePresenter: NoticePresenter = ServiceLocator.noticePresenter
+    private let analytics: Analytics = ServiceLocator.analytics
+
+    private var cancellables = Set<AnyCancellable>()
+
+    /// Start presenting notices for the given fulfillment process.
+    ///
+    /// Undo and retry options will be presented to the user too.
+    func present(process: OrderFulfillmentUseCase.FulfillmentProcess) {
+        displayOptimisticFulfillmentNotice(process)
+        observe(fulfillmentProcess: process)
+    }
+
+    /// Observe the given process and display error notices if needed.
+    private func observe(fulfillmentProcess: OrderFulfillmentUseCase.FulfillmentProcess) {
+        var cancellable: AnyCancellable = AnyCancellable { }
+        cancellable = fulfillmentProcess.result.sink { completion in
+            if case .failure(let fulfillmentError) = completion {
+                self.displayFulfillmentErrorNotice(error: fulfillmentError)
+            }
+
+            cancellable.cancel()
```

Comment:
Oh, good point. The `cancel()` is not necessary I think because `Future` is like `Single`. When it completes, it doesn't send any values anymore. So the `Subscription` effectively ends. 

The important thing is to remove the `AnyCancellable` from memory. Based on my test, the `Subscriber.Sink` and `AnyCancellable` were still kept in memory because they're stored in `self.cancellables` (`Set<>`). I added a removal in f0bd0e50d6070cdea2d2b400021214e1df7ba4d1. 

Thanks for pointing that out!

I also noticed the `OrderFulfillmentNoticePresenter` still kept in memory but I can't figure out why. I'll fix it in another PR.